node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/QuestMetricsDb.java

```diff
@@ -196,8 +199,12 @@ public void gc() {
 
     @Override
     public void close() {
-        if (engine != null)
-            engine.close();
+        synchronized (clusterTable.writeLock) {
+            CairoEngine myEngine = engine.getAndSet(null);
```

Comment:
This seems a little too complicated to me, why not just have an atomic boolean "closed"?
But we still need to check for it everywhere - do we actually have any problems with this?

Reply:
We have a issue that shows up in JNI or very lowlevel(UnSafe) java code that we can not explain.
Then it is at least good to verify that we are not accessing objects after they are invalidated.

Using an AtomicBoolean is fine and add checks. This was just a suggestion. There might be better ways to do this.
I do not know the threading model here so I can not say what is the best approach.
But detecting use after close, and closing down the SqlCompilers are minimum I think.
I see that the SqlExecutionContext is Closable too, but calling close() on it seems dangerous as it seems to close a shared resource(MessageBus).